data/vibe/data/serialization.d

```diff
@@ -597,14 +617,14 @@ private template deserializeValueImpl(Serializer, alias Policy) {
 		static if (isPolicySerializable!(Policy, T)) {
 			alias CustomType = typeof(Policy!T.toRepresentation(T.init));
 			return Policy!T.fromRepresentation(ser.deserializeValue!(CustomType, ATTRIBUTES));
+		} else static if (Serializer.isSupportedValueType!T) {
```

Comment:
I wonder whether `isSupportedValueType!T` wrongly returns true for some of the serializers (e.g. due to `is(T : long)` or a similar condition).